handlers/officerDeltaHandler.go

```diff
@@ -1,22 +1,45 @@
 package handlers
 
 import (
+	"github.com/companieshouse/chs-delta-api/helpers"
+	"github.com/companieshouse/chs-delta-api/services"
 	"net/http"
 )
 
-// OfficerDeltaHandler offers a handler by which to publish a message onto a kafka topic.
+const (
+	officersTopic = "officers-delta"
+)
+
+// OfficerDeltaHandler offers a handler by which to publish an office-delta onto the officer-delta kafka topic.
 type OfficerDeltaHandler struct {
+	KSvc services.KafkaService
+	h helpers.Helper
 }
 
 // NewOfficerDeltaHandler returns an OfficerDeltaHandler.
-func NewOfficerDeltaHandler() *OfficerDeltaHandler {
-	return &OfficerDeltaHandler{}
+func NewOfficerDeltaHandler(kSvc services.KafkaService, h helpers.Helper) *OfficerDeltaHandler {
+	return &OfficerDeltaHandler{KSvc: kSvc, h: h}
 }
 
 // ServeHTTP accepts an incoming OfficerDelta request via a POST method, validates it
-// and then passes it to a Kafka service for further processing. If errors are encountered
-// then they will be returned via the ResponseWriter.
-func (kp *OfficerDeltaHandler) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
+// and then passes it to a Kafka service for further processing along with an officers-delta topic. If errors are
+// encountered then they will be returned via the ResponseWriter.
+func (kp *OfficerDeltaHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	// Get request body.
+	data, err := kp.h.GetDataFromRequest(r)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
Should this be a BadRequest? I don't think the request body is being validated here, so I think we should be returning an `InternalServerError`